src/Swift.Runtime/src/Swift/SwiftString.cs

```diff
@@ -0,0 +1,170 @@
+    /// </summary>
+    public SwiftString(string str)
+    {
+        byte[] utf8Bytes = Encoding.UTF8.GetBytes(str);
+        unsafe
+        {
+            fixed (byte* utf8BytesPtr = utf8Bytes)
+            {
+                _payload = Create(utf8BytesPtr, utf8Bytes.Length, 1);
+            }
+        }
+    }
+
+    /// <summary>
+    /// Gets the length of string.
+    /// </summary>
+    public int Length => (int)GetLength(_payload);
+
+    /// <summary>
+    /// Converts the SwiftString to a C# string.
+    /// </summary>
+    public override string ToString()
+    {
+        unsafe
+        {
+            var elementType = TypeMetadata.GetTypeMetadataOrThrow<byte>();
+            var resultType = TypeMetadata.GetTypeMetadataOrThrow<long>();
+
+            var len = Length;
+            if (len <= 0)
+                return string.Empty;
+
+            string? str = null;
+
+            var arr = GetUtf8ContiguousArray(_payload);
+            GCHandle handle = GCHandle.Alloc(str, GCHandleType.Pinned);
+            try
+            {
+                WithUnsafeBytes(bytes =>
+                    {
+                        unsafe
+                        {
+                            str = Encoding.UTF8.GetString((byte*)bytes, len);
+                            return IntPtr.Zero;
+                        }
+                    }, IntPtr.Zero, arr, elementType, resultType);
+            }
+            finally
+            {
+                handle.Free();
+            }
+
+            return str!;
+        }
+    }
+
+    public unsafe delegate IntPtr CallbackDelegate(IntPtr param);
+
+    [UnmanagedCallConv(CallConvs = new Type[] { typeof(CallConvSwift) })]
+    [DllImport(KnownLibraries.SwiftCore, EntryPoint = "$sSSMa")]
+    public static extern TypeMetadata PInvoke_getMetadata();
+
+    [UnmanagedCallConv(CallConvs = new Type[] { typeof(CallConvSwift) })]
+    [DllImport(KnownLibraries.SwiftCore, CharSet = CharSet.Unicode, EntryPoint = "$sSS21_builtinStringLiteral17utf8CodeUnitCount7isASCIISSBp_BwBi1_tcfC")]
+    public static unsafe extern Data Create(byte* str, long len, byte flag);
+
+    [UnmanagedCallConv(CallConvs = new Type[] { typeof(CallConvSwift) })]
+    [DllImport(KnownLibraries.SwiftCore, EntryPoint = "$sSS5countSivg")]
+    public static extern long GetLength(Data str);
+
+    [UnmanagedCallConv(CallConvs = new Type[] { typeof(CallConvSwift) })]
+    [DllImport(KnownLibraries.SwiftCore, EntryPoint = "$sSS11utf8CStrings15ContiguousArrayVys4Int8VGvg")]
+    public static unsafe extern IntPtr GetUtf8ContiguousArray(Data str);
+
+    [UnmanagedCallConv(CallConvs = new Type[] { typeof(CallConvSwift) })]
+    [DllImport(KnownLibraries.SwiftCore, EntryPoint = "$ss15ContiguousArrayV15withUnsafeBytesyqd__qd__SWKXEKlF")]
+    public static extern unsafe IntPtr WithUnsafeBytes(CallbackDelegate callback, IntPtr context, IntPtr contiguousArray, TypeMetadata elementType, TypeMetadata resultType);
```

Comment:
I'm curious, is there some doc which described these functions?
It would be really nice to link to that doc here.

Reply:
Added:
 - https://developer.apple.com/documentation/swift/string/utf8cstring
 - https://developer.apple.com/documentation/swift/contiguousarray/withunsafebytes(_:)